File: code/java/ExchangeDataFeedSnapshotAPISymbolList/v1/src/main/java/com/factset/sdk/ExchangeDataFeedSnapshotAPISymbolList/models/DFSnapshotResponse.java
/*
 * Exchange DataFeed Snapshot
 * FactSet’s Exchange DataFeed Snapshot API provides cost-effective access to real-time and delayed global exchange data. Proprietary technology normalizes over 200 global exchanges and 150+ data fields. Asset types integrated include equities, futures, options, warrants, fixed income, mutual funds, ETFs, indices, commodities, and FX rates. <p>Cutting-edge technology ensures reliability and provides scalability that allow applications to request multiple items at a time. To simplify client-side development an entire response can be placed in a matrix or table for effortless integration into internal and external applications. Using specified output formats (CSV, XML, JSON) receive all standard fields by default or customize the list based on specific needs.</p></p>Below are the current hosts:</p><p>Production: api.factset.com<p>Sandbox: api-sandbox.factset.com</p>
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: api@factset.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package com.factset.sdk.ExchangeDataFeedSnapshotAPISymbolList.models;

import java.util.Objects;
import java.util.Arrays;
import java.util.Map;
import java.util.HashMap;
import com.factset.sdk.ExchangeDataFeedSnapshotAPISymbolList.models.Fields;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonTypeName;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.Serializable;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.factset.sdk.ExchangeDataFeedSnapshotAPISymbolList.JSON;


/**
 * sample response
 */
@ApiModel(description = "sample response")
@JsonPropertyOrder({
  DFSnapshotResponse.JSON_PROPERTY_ERROR,
  DFSnapshotResponse.JSON_PROPERTY_SYMBOL1,
  DFSnapshotResponse.JSON_PROPERTY_HOST,
  DFSnapshotResponse.JSON_PROPERTY_SYMBOL2
})
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen")
public class DFSnapshotResponse implements Serializable {
  private static final long serialVersionUID = 1L;

  public static final String JSON_PROPERTY_ERROR = "Error";
  private Integer error;

  public static final String JSON_PROPERTY_SYMBOL1 = "symbol1";
  private java.util.List<Fields> symbol1 = null;

  public static final String JSON_PROPERTY_HOST = "Host";
  private String host;

  public static final String JSON_PROPERTY_SYMBOL2 = "symbol2";
  private String symbol2;


  public DFSnapshotResponse error(Integer error) {
    this.error = error;
    return this;
  }

   /**
   * error value
   * @return error
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(example = "0", value = "error value")
  @JsonProperty(JSON_PROPERTY_ERROR)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public Integer getError() {
    return error;
  }


  @JsonProperty(JSON_PROPERTY_ERROR)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setError(Integer error) {
    this.error = error;
  }


  public DFSnapshotResponse symbol1(java.util.List<Fields> symbol1) {
    this.symbol1 = symbol1;
    return this;
  }

  public DFSnapshotResponse addSymbol1Item(Fields symbol1Item) {
    if (this.symbol1 == null) {
      this.symbol1 = new java.util.ArrayList<>();
    }
    this.symbol1.add(symbol1Item);
    return this;
  }

   /**
   * first symbol from &#x60;ids&#x60; or &#x60;chain&#x60;
   * @return symbol1
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "first symbol from `ids` or `chain`")
  @JsonProperty(JSON_PROPERTY_SYMBOL1)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public java.util.List<Fields> getSymbol1() {
    return symbol1;
  }


  @JsonProperty(JSON_PROPERTY_SYMBOL1)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setSymbol1(java.util.List<Fields> symbol1) {
    this.symbol1 = symbol1;
  }


  public DFSnapshotResponse host(String host) {
    this.host = host;
    return this;
  }

   /**
   * host name
   * @return host
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(example = "FactSet", value = "host name")
  @JsonProperty(JSON_PROPERTY_HOST)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getHost() {
    return host;
  }


  @JsonProperty(JSON_PROPERTY_HOST)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setHost(String host) {
    this.host = host;
  }


  public DFSnapshotResponse symbol2(String symbol2) {
    this.symbol2 = symbol2;
    return this;
  }

   /**
   * second symbol from &#x60;ids&#x60; or &#x60;chain&#x60; list
   * @return symbol2
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "second symbol from `ids` or `chain` list")
  @JsonProperty(JSON_PROPERTY_SYMBOL2)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)

  public String getSymbol2() {
    return symbol2;
  }


  @JsonProperty(JSON_PROPERTY_SYMBOL2)
  @JsonInclude(value = JsonInclude.Include.USE_DEFAULTS)
  public void setSymbol2(String symbol2) {
    this.symbol2 = symbol2;
  }


  /**
   * Return true if this DFSnapshot_Response object is equal to o.
   */
  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DFSnapshotResponse dfSnapshotResponse = (DFSnapshotResponse) o;
    return Objects.equals(this.error, dfSnapshotResponse.error) &&
        Objects.equals(this.symbol1, dfSnapshotResponse.symbol1) &&
        Objects.equals(this.host, dfSnapshotResponse.host) &&
        Objects.equals(this.symbol2, dfSnapshotResponse.symbol2);
  }

  @Override
  public int hashCode() {
    return Objects.hash(error, symbol1, host, symbol2);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DFSnapshotResponse {\n");
    sb.append("    error: ").append(toIndentedString(error)).append("\n");
    sb.append("    symbol1: ").append(toIndentedString(symbol1)).append("\n");
    sb.append("    host: ").append(toIndentedString(host)).append("\n");
    sb.append("    symbol2: ").append(toIndentedString(symbol2)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: code/dotnet/ExchangeDataFeedSnapshotAPISymbolList/v1/src/FactSet.SDK.ExchangeDataFeedSnapshotAPISymbolList/Model/DFSnapshotResponse.cs
/*
 * Exchange DataFeed Snapshot
 *
 * FactSet’s Exchange DataFeed Snapshot API provides cost-effective access to real-time and delayed global exchange data. Proprietary technology normalizes over 200 global exchanges and 150+ data fields. Asset types integrated include equities, futures, options, warrants, fixed income, mutual funds, ETFs, indices, commodities, and FX rates. <p>Cutting-edge technology ensures reliability and provides scalability that allow applications to request multiple items at a time. To simplify client-side development an entire response can be placed in a matrix or table for effortless integration into internal and external applications. Using specified output formats (CSV, XML, JSON) receive all standard fields by default or customize the list based on specific needs.</p></p>Below are the current hosts:</p><p>Production: api.factset.com<p>Sandbox: api-sandbox.factset.com</p>
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: api@factset.com
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = FactSet.SDK.ExchangeDataFeedSnapshotAPISymbolList.Client.OpenAPIDateConverter;

namespace FactSet.SDK.ExchangeDataFeedSnapshotAPISymbolList.Model
{
    /// <summary>
    /// sample response
    /// </summary>
    [DataContract(Name = "DFSnapshot_Response")]
    public partial class DFSnapshotResponse : IEquatable<DFSnapshotResponse>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="DFSnapshotResponse" /> class.
        /// </summary>
        /// <param name="error">error value.</param>
        /// <param name="symbol1">first symbol from &#x60;ids&#x60; or &#x60;chain&#x60;.</param>
        /// <param name="host">host name.</param>
        /// <param name="symbol2">second symbol from &#x60;ids&#x60; or &#x60;chain&#x60; list.</param>
        public DFSnapshotResponse(int error = default(int), List<Fields> symbol1 = default(List<Fields>), string host = default(string), string symbol2 = default(string))
        {
            this.Error = error;
            this.Symbol1 = symbol1;
            this.Host = host;
            this.Symbol2 = symbol2;
        }

        /// <summary>
        /// error value
        /// </summary>
        /// <value>error value</value>
        [DataMember(Name = "Error", EmitDefaultValue = false)]
        public int Error { get; set; }

        /// <summary>
        /// first symbol from &#x60;ids&#x60; or &#x60;chain&#x60;
        /// </summary>
        /// <value>first symbol from &#x60;ids&#x60; or &#x60;chain&#x60;</value>
        [DataMember(Name = "symbol1", EmitDefaultValue = false)]
        public List<Fields> Symbol1 { get; set; }

        /// <summary>
        /// host name
        /// </summary>
        /// <value>host name</value>
        [DataMember(Name = "Host", EmitDefaultValue = false)]
        public string Host { get; set; }

        /// <summary>
        /// second symbol from &#x60;ids&#x60; or &#x60;chain&#x60; list
        /// </summary>
        /// <value>second symbol from &#x60;ids&#x60; or &#x60;chain&#x60; list</value>
        [DataMember(Name = "symbol2", EmitDefaultValue = false)]
        public string Symbol2 { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class DFSnapshotResponse {\n");
            sb.Append("  Error: ").Append(Error).Append("\n");
            sb.Append("  Symbol1: ").Append(Symbol1).Append("\n");
            sb.Append("  Host: ").Append(Host).Append("\n");
            sb.Append("  Symbol2: ").Append(Symbol2).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as DFSnapshotResponse);
        }

        /// <summary>
        /// Returns true if DFSnapshotResponse instances are equal
        /// </summary>
        /// <param name="input">Instance of DFSnapshotResponse to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(DFSnapshotResponse input)
        {
            if (input == null)
            {
                return false;
            }
            return 
                (
                    this.Error == input.Error ||
                    this.Error.Equals(input.Error)
                ) && 
                (
                    this.Symbol1 == input.Symbol1 ||
                    this.Symbol1 != null &&
                    input.Symbol1 != null &&
                    this.Symbol1.SequenceEqual(input.Symbol1)
                ) && 
                (
                    this.Host == input.Host ||
                    (this.Host != null &&
                    this.Host.Equals(input.Host))
                ) && 
                (
                    this.Symbol2 == input.Symbol2 ||
                    (this.Symbol2 != null &&
                    this.Symbol2.Equals(input.Symbol2))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                hashCode = (hashCode * 59) + this.Error.GetHashCode();
                if (this.Symbol1 != null)
                {
                    hashCode = (hashCode * 59) + this.Symbol1.GetHashCode();
                }
                if (this.Host != null)
                {
                    hashCode = (hashCode * 59) + this.Host.GetHashCode();
                }
                if (this.Symbol2 != null)
                {
                    hashCode = (hashCode * 59) + this.Symbol2.GetHashCode();
                }
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        public IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}
